pkgs/apps/ambertools/default.nix

```diff
@@ -0,0 +1,143 @@
+{ buildPythonPackage
+, lib
+, requireFile
+, makeWrapper
+  # Python dependencies
+, numpy
+, scipy
+, matplotlib
+, setuptools
+  # Native dependencies
+, gfortran
+, cmake
+, readline
+, perl
+, flex
+, bison
+, zlib
+, boost
+, netcdf
+, netcdffortran
+, fftw
+, blas
+, lapack
+, protobuf
+, plumed
+, apbs
+, arpack
+, runtimeShell
+}:
+
+buildPythonPackage rec {
+  pname = "AmberTools";
+  version = "24";
+
+  src = requireFile {
+    name = "AmberTools${version}.tar.bz2";
+    sha256 = "sha256-UvtPszcKibfOc4otw+UTwvwZQ/3ktDgYRtnnXMSNhA8=";
+    url = "https://ambermd.org/AmberTools.php";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    gfortran
+    flex
+    bison
+    makeWrapper
+  ];
+
+  buildInputs = [
+    zlib
+    boost
+    blas
+    lapack
+    netcdffortran
+    fftw
+    protobuf
+    plumed
+    arpack
+    apbs
+    readline
+  ];
+
+  format = "other";
+
+  cmakeFlags = [
+    "-DCOMPILER=AUTO"
+    "-DDOWNLOAD_MINICONDA=OFF"
+    "-DOPENMP=ON"
+    "-DTRUST_SYSTEM_LIBS=ON"
+  ];
+
+  propagatedBuildInputs = [
+    perl
+  ];
+
+  dependencies = [
+    numpy
+    scipy
+    matplotlib
+    setuptools
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    make -j $NIX_BUILD_CORES
+
+    runHook postBuild
+  '';
+
+  installPhase =
+    let
+
+      wrongBash = [
+        "am1bcc"
+        "antechamber"
+        "atomtype"
+        "bondtype"
+        "espgen"
+        "match"
+        "match_atomname"
+        "parmcal"
+        "parmchk2"
+        "prepgen"
+        "reduce"
+        "residuegen"
+        "respgen"
+        "XrayPrep"
+      ];
+    in
+    ''
+      runHook preInstall
+
+      make install
+
+      # Some scripts hardcode /bin/bash. Not only necessary as their shebang, but
+      # some also generate bash scripts with wrong shebangs.
+      for PROG in ${builtins.toString wrongBash}; do
+        substituteInPlace $out/bin/$PROG --replace-fail '#!/bin/bash' '#!${runtimeShell}'
```

Comment:
It doesn't for some reasons. I don't know why. But even if it would patch the shebangs, the hardcoded path would also need to be removed from some Python strings.